h2o-core/src/main/java/water/MemKeyed.java

```diff
@@ -0,0 +1,31 @@
+package water;
+
+public class MemKeyed<T extends MemKeyed<T>> extends Keyed<T> {
+
+  protected byte[] _mem;
+
+  public MemKeyed(Key<T> key, byte[] mem) {
+    super(key);
+    _mem = mem;
+  }
+
+  @SuppressWarnings("unchecked")
+  public final T reloadFromBytes(Key<T> k, byte[] ary) {
+    _key = k;
+    _mem = ary;
```

Comment:
Just to make sure i understand correctly, this is the key trick - simply do not materialize a new object from `_mem` but simply just keep reference (which is shared from `Value`).

Reply:
yes, precisely - that is the trick